src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,854 @@
+function (G::MatrixGroup)(x::MatElem)
+   vero, x_gap = lies_in(x,G,nothing)
+   vero || throw(ArgumentError("Element not in the group"))
+   if x_gap==nothing return MatrixGroupElem(G,x)
+   else return MatrixGroupElem(G,x,x_gap)
+   end
+end
+
+# embedding an element of type MatrixGroupElem into a group G
+function (G::MatrixGroup)(x::MatrixGroupElem)
+   if isdefined(x,:X)
+      if isdefined(x,:elm)
+         vero = lies_in(x.elm,G,x.X)[1]
+         vero || throw(ArgumentError("Element not in the group"))
+         return MatrixGroupElem(G,x.elm,x.X)
+      else
+         GAP.Globals.in(x.X, G.X) || throw(ArgumentError("Element not in the group"))
+         return MatrixGroupElem(G,x.X)
+      end
+   else
+      vero, x_gap = lies_in(x.elm,G,nothing)
+      vero || throw(ArgumentError("Element not in the group"))
+      if x_gap==nothing return MatrixGroupElem(G,x.elm)
+      else return MatrixGroupElem(G,x.elm,x_gap)
+      end
+   end
+end
+
+# embedding a nxn array into a group G
+function (G::MatrixGroup)(L::AbstractVector)
+   x = matrix(G.ring, G.deg, G.deg, L)
+   return G(x)
+end
+
+########################################################################
+#
+# Methods on elements
+#
+########################################################################
+
+# TODO: we are not currently keeping track of the parent
+function ==(x::MatrixGroupElem{S,T},y::MatrixGroupElem{S,T}) where {S,T}
+   if isdefined(x,:X) && isdefined(y,:X) return x.X==y.X
+   else return x.elm==y.elm
+   end
+end
+
+# TODO: we wish to multiply / conjugate also matrices with different parents ?
+function _prod(x::MatrixGroupElem,y::MatrixGroupElem)
+   if x.parent==y.parent
+      if isdefined(x,:X) && isdefined(y,:X) && !(isdefined(x,:elm) && isdefined(y,:elm))
+         return MatrixGroupElem(x.parent, x.X*y.X)
+      else
+         return MatrixGroupElem(x.parent, x.elm*y.elm)
+      end
+   else
+      throw(ArgumentError("Matrices not in the same group"))
+   end
+end
+# Base.:* is defined in src/Groups/GAPGroups.jl
+
+Base.:*(x::MatrixGroupElem, y::fq_nmod_mat) = x.elm*y
+Base.:*(x::fq_nmod_mat, y::MatrixGroupElem) = x*y.elm
+
+Base.:^(x::MatrixGroupElem, n::Int) = MatrixGroupElem(x.parent, x.elm^n)
+
+Base.isone(x::MatrixGroupElem) = isone(x.elm)
+
+Base.inv(x::MatrixGroupElem) = MatrixGroupElem(x.parent, inv(x.elm))
+
+# TODO: we wish to multiply / conjugate also matrices with different parents ?
+function Base.:^(x::MatrixGroupElem, y::MatrixGroupElem)
+   if x.parent==y.parent
+      if isdefined(x,:X) && isdefined(y,:X) && !(isdefined(x,:elm) && isdefined(y,:elm))
+         return MatrixGroupElem(x.parent, inv(y.X)*x.X*y.X)
+      else
+         return MatrixGroupElem(x.parent,inv(y.elm)*x.elm*y.elm)
+      end
+   else
+      throw(ArgumentError("Matrices not in the same group"))
+   end
+end
+
+comm(x::MatrixGroupElem, y::MatrixGroupElem) = inv(x)*conj(x,y)
+
+"""
+    det(x::MatrixGroupElem)
+Return the determinant of `x`.
```

Comment:
We just cleaned-up our docstrings. We now try to have an empty line between the signature and the docstring:
```suggestion
    det(x::MatrixGroupElem)
    
Return the determinant of `x`.
```